hazelcast-client/src/test/java/com/hazelcast/client/spi/ClientProxyDestroyTest.java

```diff
@@ -81,4 +85,36 @@ public void testOperationAfterDestroy() throws Exception {
         clientMap.put(1, 1);
         assertEquals(1, clientMap.get(1));
     }
+
+    @Test
+    @Category(SlowTest.class)
+    public void testRemoteProxyCreationDelegatesToClientEventually() {
+        final HazelcastClientInstanceImpl clientInstanceImpl = getHazelcastClientInstanceImpl(client);
+        assertEquals(0, clientInstanceImpl.getProxyManager().getLocalDistributedObjects().size());
+        server.getMap("map");
```

Comment:
if you use a second client instance and do this remote operation, the test can be easily duplicated by non-java clients as well.

Reply:
I will change the test accordingly.